src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -455,7 +453,13 @@ private async Task ExecuteClusterTransactionTask()
                     using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
                     using (context.OpenReadTransaction())
                     {
-                        await ExecuteClusterTransaction(context);
+                        const int batchSize = 256;
+                        var executed = await ExecuteClusterTransaction(context, batchSize: batchSize);
+                        if (executed.Count == batchSize)
```

Comment:
why do this only if there is a match? Shouldn't we do that if there is _any_ results?

Reply:
I think it matches previous behavior?

Reply:
@ppekrol that is correct.
The previous behavior didn't take thet into account.
Previously if we had 300 transactions that we didn't process, we would only process 256 of them.